Fails to build with Boost 1.72.0

Regressed by https://github.com/boostorg/gil/commit/5611bd580718
```
$ c++ --version
FreeBSD clang version 9.0.0 (tags/RELEASE_900/final 372316) (based on LLVM 9.0.0)
Target: x86_64-unknown-freebsd13.0
Thread model: posix
InstalledDir: /usr/bin
```
```c++
FAILED: GG/CMakeFiles/GiGi.dir/src/Texture.cpp.o
/usr/bin/c++  -DBINPATH=\"/usr/local/bin\" -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_GIL_IO_ENABLE_GRAY_ALPHA -DBOOST_LOG_DYN_LINK -DBOOST_LOG_NO_LIB -DENABLE_BINRELOC -DFONS_USE_FREETYPE -DFREEORION_FREEBSD -DGiGi_EXPORTS -DSHAREPATH=\"/usr/local/share\" -I/path/to/freeorion/GG -IGG -isystem /usr/local/include -isystem /usr/local/include/freetype2 -O3 -DNDEBUG -fPIC   -Wall -std=c++11 -MD -MT GG/CMakeFiles/GiGi.dir/src/Texture.cpp.o -MF GG/CMakeFiles/GiGi.dir/src/Texture.cpp.o.d -o GG/CMakeFiles/GiGi.dir/src/Texture.cpp.o -c /path/to/freeorion/GG/src/Texture.cpp
/path/to/freeorion/GG/src/Texture.cpp:218:25: error: no template named 'vector4' in namespace 'boost::mpl'
    typedef boost::mpl::vector4<
            ~~~~~~~~~~~~^
In file included from /path/to/freeorion/GG/src/Texture.cpp:32:
In file included from /usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:11:
In file included from /usr/local/include/boost/gil/extension/dynamic_image/any_image_view.hpp:17:
In file included from /usr/local/include/boost/variant.hpp:17:
/usr/local/include/boost/variant/variant.hpp:2416:5: error: static_assert failed due to requirement '::boost::mpl::is_sequence<int>::value' "( ::boost::mpl::is_sequence<Types>::value )"
    BOOST_STATIC_ASSERT(( ::boost::mpl::is_sequence<Types>::value ));
    ^                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/boost/static_assert.hpp:70:41: note: expanded from macro 'BOOST_STATIC_ASSERT'
#     define BOOST_STATIC_ASSERT( ... ) static_assert(__VA_ARGS__, #__VA_ARGS__)
                                        ^             ~~~~~~~~~~~
/usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:88:26: note: in instantiation of template class 'boost::make_variant_over<int>' requested here
class any_image : public make_variant_over<Images>::type
                         ^
/path/to/freeorion/GG/src/Texture.cpp:233:15: note: in instantiation of template class 'boost::gil::any_image<int>' requested here
    ImageType image;
              ^
In file included from /path/to/freeorion/GG/src/Texture.cpp:32:
/usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:90:58: error: no type named 'type' in 'boost::make_variant_over<int>'
    using parent_t = typename make_variant_over<Images>::type;
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
/path/to/freeorion/GG/src/Texture.cpp:233:15: note: in instantiation of template class 'boost::gil::any_image<int>' requested here
    ImageType image;
              ^
In file included from /path/to/freeorion/GG/src/Texture.cpp:32:
In file included from /usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:11:
In file included from /usr/local/include/boost/gil/extension/dynamic_image/any_image_view.hpp:12:
In file included from /usr/local/include/boost/gil/image.hpp:11:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
In file included from /usr/local/include/boost/gil/pixel.hpp:12:
In file included from /usr/local/include/boost/gil/channel.hpp:11:
In file included from /usr/local/include/boost/gil/utilities.hpp:11:
In file included from /usr/local/include/boost/gil/detail/mp11.hpp:12:
In file included from /usr/local/include/boost/mp11.hpp:11:
/usr/local/include/boost/mp11/list.hpp:45:19: error: no type named 'type' in 'boost::mp11::detail::mp_size_impl<int>'
template<class L> using mp_size = typename detail::mp_size_impl<L>::type;
                  ^~~~~
/usr/local/include/boost/mp11/algorithm.hpp:121:50: note: in instantiation of template type alias 'mp_size' requested here
template<template<class...> class F, class... L> using mp_transform = typename mp_if<mp_same<mp_size<L>...>, detail::mp_transform_impl<F, L...>, detail::list_size_mismatch>::type;
                                                 ^
/usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:33:1: note: in instantiation of template type alias 'mp_transform' requested here
using images_get_views_t = mp11::mp_transform<get_view_t, Images>;
^
/usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:92:43: note: in instantiation of template type alias 'images_get_views_t' requested here
    using view_t = any_image_view<detail::images_get_views_t<Images>>;
                                          ^
/path/to/freeorion/GG/src/Texture.cpp:233:15: note: in instantiation of template class 'boost::gil::any_image<int>' requested here
    ImageType image;
              ^
In file included from /path/to/freeorion/GG/src/Texture.cpp:32:
In file included from /usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:11:
In file included from /usr/local/include/boost/gil/extension/dynamic_image/any_image_view.hpp:12:
In file included from /usr/local/include/boost/gil/image.hpp:11:
In file included from /usr/local/include/boost/gil/algorithm.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_iterator.hpp:11:
In file included from /usr/local/include/boost/gil/bit_aligned_pixel_reference.hpp:12:
In file included from /usr/local/include/boost/gil/pixel.hpp:12:
In file included from /usr/local/include/boost/gil/channel.hpp:11:
In file included from /usr/local/include/boost/gil/utilities.hpp:11:
In file included from /usr/local/include/boost/gil/detail/mp11.hpp:12:
In file included from /usr/local/include/boost/mp11.hpp:11:
/usr/local/include/boost/mp11/list.hpp:45:19: error: no type named 'type' in 'boost::mp11::detail::mp_size_impl<int>'
template<class L> using mp_size = typename detail::mp_size_impl<L>::type;
                  ^~~~~
/usr/local/include/boost/mp11/algorithm.hpp:121:50: note: in instantiation of template type alias 'mp_size' requested here
template<template<class...> class F, class... L> using mp_transform = typename mp_if<mp_same<mp_size<L>...>, detail::mp_transform_impl<F, L...>, detail::list_size_mismatch>::type;
                                                 ^
/usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:39:1: note: in instantiation of template type alias 'mp_transform' requested here
using images_get_const_views_t = mp11::mp_transform<get_const_view_t, Images>;
^
/usr/local/include/boost/gil/extension/dynamic_image/any_image.hpp:93:49: note: in instantiation of template type alias 'images_get_const_views_t' requested here
    using const_view_t = any_image_view<detail::images_get_const_views_t<Images>>;
                                                ^
/path/to/freeorion/GG/src/Texture.cpp:233:15: note: in instantiation of template class 'boost::gil::any_image<int>' requested here
    ImageType image;
              ^
5 errors generated.
```

--- a/GG/CMakeLists.txt
+++ b/GG/CMakeLists.txt
@@ -153,6 +153,7 @@ target_compile_definitions(GiGi
         # instead of our hacky gilext code.
         $<$<VERSION_LESS:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.68>:GIGI_CONFIG_USE_OLD_IMPLEMENTATION_OF_GIL_PNG_IO>
         $<$<VERSION_GREATER:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.67>:BOOST_GIL_IO_ENABLE_GRAY_ALPHA>
+        $<$<VERSION_GREATER:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.71>:BOOST_GIL_USES_MP11>
 )
 
 target_include_directories(GiGi SYSTEM

--- a/GG/src/Texture.cpp
+++ b/GG/src/Texture.cpp
@@ -5,12 +5,12 @@
    modify it under the terms of the GNU Lesser General Public License
    as published by the Free Software Foundation; either version 2.1
    of the License, or (at your option) any later version.
-   
+
    This library is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    Lesser General Public License for more details.
-    
+
    You should have received a copy of the GNU Lesser General Public
    License along with this library; if not, write to the Free
    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
@@ -18,7 +18,7 @@
 
    If you do not wish to comply with the terms of the LGPL please
    contact the author as other terms are available for a fee.
-    
+
    Zach Laine
    whatwasthataddress@gmail.com */
 
@@ -215,7 +215,11 @@ void Texture::Load(const boost::filesystem::path& path, bool mipmap/* = false*/)
     BOOST_STATIC_ASSERT((sizeof(gil::rgb8_pixel_t) == 3));
     BOOST_STATIC_ASSERT((sizeof(gil::rgba8_pixel_t) == 4));
 
+#ifdef BOOST_GIL_USES_MP11
+    typedef boost::mp11::mp_list<
+#else
     typedef boost::mpl::vector4<
+#endif
         gil::gray8_image_t,
         gil::gray_alpha8_image_t,
         gil::rgb8_image_t,

